src/Support/ArrayHelpers.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Roave\ApiCompare\Support;
+
+use Assert\Assert;
+use function in_array;
+
+/**
+ * @internal this is a support class of this library, and should NOT be used outside of it
+ */
+final class ArrayHelpers
+{
+    /**
+     * Yes, this is just a very pedantic version of `in_array()`, written to avoid mutations and
+     * designed to throw an exception if `$arrayOfStrings` is not a `string[]` as requested.
+     *
+     * @param string[] $arrayOfStrings
+     *
+     * @throws \InvalidArgumentException
+     */
+    public static function stringArrayContainsString(string $value, array $arrayOfStrings) : bool
+    {
+        Assert::that($arrayOfStrings)->all()->string();
```

Comment:
Really not sure we need this... seems to just be avoiding false positives; don't know specifically what mutations phpstan performs, but adding an integer onto this array shouldn't be a problem if the expected value is still in there (because we (were) using "strict" mode)... Think this conversation should be moved to a discussion upstream in phpstan, not solved with a wrapper IMO